errbot/plugin_wizard.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+
+import configparser
+import os
+import re
+import requests
+import sys
+
+from errbot import PY2
+
+if PY2:
+    input = raw_input
+
+
+def new_plugin_wizard(directory=None):
+    """
+    Start the wizard to create a new plugin in the current working directory.
+    """
+    if directory is None:
+        print("This wizard will create a new plugin for you in the current directory.")
+        directory = os.getcwd()
+    else:
+        print("This wizard will create a new plugin for you in '%s'." % directory)
+
+    if not os.path.isdir(directory):
+        print("Error: The path '%s' does not exist or is not a directory." % directory)
+        sys.exit(1)
+
+    try:
+        plugin_code = download_plugin_skeleton()
+    except requests.HTTPError as e:
+        print(
+            "Error: I couldn't download the plugin skeleton from GitHub "
+            "which is needed to write out a new plugin file for you."
+        )
+        print("Details: {!s}".format(e))
+        sys.exit(1)
+
+    name = ask(
+        "What should the name of your new plugin be?",
+        validation_regex=r'^[a-zA-Z][a-zA-Z0-9 _-]*$'
```

Comment:
I avoid those unless it becomes quite unreadable otherwise. I find I tend to forget which character ranges are included exactly in which `\x` set and I imagine other people suffer from the same thing.

Considering the validation is also printed on incorrect input, I think this is easier for people to understand.